File: java-play-framework/app/apimodels/Bot.java
package apimodels;

import apimodels.BasicBotBase;
import apimodels.BotAllOf;
import com.fasterxml.jackson.annotation.*;
import java.util.Set;
import javax.validation.*;
import java.util.Objects;
import javax.validation.constraints.*;
/**
 * Bot
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2021-08-08T21:11:38.527546Z[Etc/UTC]")
@SuppressWarnings({"UnusedReturnValue", "WeakerAccess"})
public class Bot   {
  @JsonProperty("user_id")
  
  private Object userId = null;

  @JsonProperty("full_name")
  
  private Object fullName = null;

  @JsonProperty("api_key")
  
  private Object apiKey = null;

  @JsonProperty("default_sending_stream")
  
  private Object defaultSendingStream = null;

  @JsonProperty("default_events_register_stream")
  
  private Object defaultEventsRegisterStream = null;

  @JsonProperty("default_all_public_streams")
  
  private Object defaultAllPublicStreams = null;

  @JsonProperty("avatar_url")
  
  private Object avatarUrl = null;

  @JsonProperty("owner_id")
  
  private Object ownerId = null;

  @JsonProperty("services")
  
  private Object services = null;

  @JsonProperty("email")
  
  private String email;

  @JsonProperty("bot_type")
  
  private Integer botType;

  @JsonProperty("is_active")
  
  private Boolean isActive;

  public Bot userId(Object userId) {
    this.userId = userId;
    return this;
  }

   /**
   * Get userId
   * @return userId
  **/
  public Object getUserId() {
    return userId;
  }

  public void setUserId(Object userId) {
    this.userId = userId;
  }

  public Bot fullName(Object fullName) {
    this.fullName = fullName;
    return this;
  }

   /**
   * Get fullName
   * @return fullName
  **/
  public Object getFullName() {
    return fullName;
  }

  public void setFullName(Object fullName) {
    this.fullName = fullName;
  }

  public Bot apiKey(Object apiKey) {
    this.apiKey = apiKey;
    return this;
  }

   /**
   * Get apiKey
   * @return apiKey
  **/
  public Object getApiKey() {
    return apiKey;
  }

  public void setApiKey(Object apiKey) {
    this.apiKey = apiKey;
  }

  public Bot defaultSendingStream(Object defaultSendingStream) {
    this.defaultSendingStream = defaultSendingStream;
    return this;
  }

   /**
   * Get defaultSendingStream
   * @return defaultSendingStream
  **/
  public Object getDefaultSendingStream() {
    return defaultSendingStream;
  }

  public void setDefaultSendingStream(Object defaultSendingStream) {
    this.defaultSendingStream = defaultSendingStream;
  }

  public Bot defaultEventsRegisterStream(Object defaultEventsRegisterStream) {
    this.defaultEventsRegisterStream = defaultEventsRegisterStream;
    return this;
  }

   /**
   * Get defaultEventsRegisterStream
   * @return defaultEventsRegisterStream
  **/
  public Object getDefaultEventsRegisterStream() {
    return defaultEventsRegisterStream;
  }

  public void setDefaultEventsRegisterStream(Object defaultEventsRegisterStream) {
    this.defaultEventsRegisterStream = defaultEventsRegisterStream;
  }

  public Bot defaultAllPublicStreams(Object defaultAllPublicStreams) {
    this.defaultAllPublicStreams = defaultAllPublicStreams;
    return this;
  }

   /**
   * Get defaultAllPublicStreams
   * @return defaultAllPublicStreams
  **/
  public Object getDefaultAllPublicStreams() {
    return defaultAllPublicStreams;
  }

  public void setDefaultAllPublicStreams(Object defaultAllPublicStreams) {
    this.defaultAllPublicStreams = defaultAllPublicStreams;
  }

  public Bot avatarUrl(Object avatarUrl) {
    this.avatarUrl = avatarUrl;
    return this;
  }

   /**
   * Get avatarUrl
   * @return avatarUrl
  **/
  public Object getAvatarUrl() {
    return avatarUrl;
  }

  public void setAvatarUrl(Object avatarUrl) {
    this.avatarUrl = avatarUrl;
  }

  public Bot ownerId(Object ownerId) {
    this.ownerId = ownerId;
    return this;
  }

   /**
   * Get ownerId
   * @return ownerId
  **/
  public Object getOwnerId() {
    return ownerId;
  }

  public void setOwnerId(Object ownerId) {
    this.ownerId = ownerId;
  }

  public Bot services(Object services) {
    this.services = services;
    return this;
  }

   /**
   * Get services
   * @return services
  **/
  public Object getServices() {
    return services;
  }

  public void setServices(Object services) {
    this.services = services;
  }

  public Bot email(String email) {
    this.email = email;
    return this;
  }

   /**
   * The email of the bot. 
   * @return email
  **/
  public String getEmail() {
    return email;
  }

  public void setEmail(String email) {
    this.email = email;
  }

  public Bot botType(Integer botType) {
    this.botType = botType;
    return this;
  }

   /**
   * An integer describing the type of bot: * `1` for a `Generic` bot. * `2` for an `Incoming webhook` bot. * `3` for an `Outgoing webhook` bot. * `4` for an `Embedded` bot. 
   * @return botType
  **/
  public Integer getBotType() {
    return botType;
  }

  public void setBotType(Integer botType) {
    this.botType = botType;
  }

  public Bot isActive(Boolean isActive) {
    this.isActive = isActive;
    return this;
  }

   /**
   * A boolean describing whether the user account has been deactivated. 
   * @return isActive
  **/
  public Boolean getIsActive() {
    return isActive;
  }

  public void setIsActive(Boolean isActive) {
    this.isActive = isActive;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Bot bot = (Bot) o;
    return Objects.equals(userId, bot.userId) &&
        Objects.equals(fullName, bot.fullName) &&
        Objects.equals(apiKey, bot.apiKey) &&
        Objects.equals(defaultSendingStream, bot.defaultSendingStream) &&
        Objects.equals(defaultEventsRegisterStream, bot.defaultEventsRegisterStream) &&
        Objects.equals(defaultAllPublicStreams, bot.defaultAllPublicStreams) &&
        Objects.equals(avatarUrl, bot.avatarUrl) &&
        Objects.equals(ownerId, bot.ownerId) &&
        Objects.equals(services, bot.services) &&
        Objects.equals(email, bot.email) &&
        Objects.equals(botType, bot.botType) &&
        Objects.equals(isActive, bot.isActive);
  }

  @Override
  public int hashCode() {
    return Objects.hash(userId, fullName, apiKey, defaultSendingStream, defaultEventsRegisterStream, defaultAllPublicStreams, avatarUrl, ownerId, services, email, botType, isActive);
  }

  @SuppressWarnings("StringBufferReplaceableByString")
  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Bot {\n");
    
    sb.append("    userId: ").append(toIndentedString(userId)).append("\n");
    sb.append("    fullName: ").append(toIndentedString(fullName)).append("\n");
    sb.append("    apiKey: ").append(toIndentedString(apiKey)).append("\n");
    sb.append("    defaultSendingStream: ").append(toIndentedString(defaultSendingStream)).append("\n");
    sb.append("    defaultEventsRegisterStream: ").append(toIndentedString(defaultEventsRegisterStream)).append("\n");
    sb.append("    defaultAllPublicStreams: ").append(toIndentedString(defaultAllPublicStreams)).append("\n");
    sb.append("    avatarUrl: ").append(toIndentedString(avatarUrl)).append("\n");
    sb.append("    ownerId: ").append(toIndentedString(ownerId)).append("\n");
    sb.append("    services: ").append(toIndentedString(services)).append("\n");
    sb.append("    email: ").append(toIndentedString(email)).append("\n");
    sb.append("    botType: ").append(toIndentedString(botType)).append("\n");
    sb.append("    isActive: ").append(toIndentedString(isActive)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: cpp-tiny/lib/Models/Bot.cpp


#include "Bot.h"

using namespace Tiny;

Bot::Bot()
{
	user_id = null;
	full_name = null;
	api_key = null;
	default_sending_stream = null;
	default_events_register_stream = null;
	default_all_public_streams = null;
	avatar_url = null;
	owner_id = null;
	services = null;
	email = std::string();
	bot_type = int(0);
	is_active = bool(false);
}

Bot::Bot(std::string jsonString)
{
	this->fromJson(jsonString);
}

Bot::~Bot()
{

}

void
Bot::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *user_idKey = "user_id";
    
    if(object.has_key(user_idKey)) 
    {
        bourne::json value = object[user_idKey];


        

        AnyType* obj = &user_id;
		obj->fromJson(value.dump());

    }

    const char *full_nameKey = "full_name";
    
    if(object.has_key(full_nameKey)) 
    {
        bourne::json value = object[full_nameKey];


        

        AnyType* obj = &full_name;
		obj->fromJson(value.dump());

    }

    const char *api_keyKey = "api_key";
    
    if(object.has_key(api_keyKey)) 
    {
        bourne::json value = object[api_keyKey];


        

        AnyType* obj = &api_key;
		obj->fromJson(value.dump());

    }

    const char *default_sending_streamKey = "default_sending_stream";
    
    if(object.has_key(default_sending_streamKey)) 
    {
        bourne::json value = object[default_sending_streamKey];


        

        AnyType* obj = &default_sending_stream;
		obj->fromJson(value.dump());

    }

    const char *default_events_register_streamKey = "default_events_register_stream";
    
    if(object.has_key(default_events_register_streamKey)) 
    {
        bourne::json value = object[default_events_register_streamKey];


        

        AnyType* obj = &default_events_register_stream;
		obj->fromJson(value.dump());

    }

    const char *default_all_public_streamsKey = "default_all_public_streams";
    
    if(object.has_key(default_all_public_streamsKey)) 
    {
        bourne::json value = object[default_all_public_streamsKey];


        

        AnyType* obj = &default_all_public_streams;
		obj->fromJson(value.dump());

    }

    const char *avatar_urlKey = "avatar_url";
    
    if(object.has_key(avatar_urlKey)) 
    {
        bourne::json value = object[avatar_urlKey];


        

        AnyType* obj = &avatar_url;
		obj->fromJson(value.dump());

    }

    const char *owner_idKey = "owner_id";
    
    if(object.has_key(owner_idKey)) 
    {
        bourne::json value = object[owner_idKey];


        

        AnyType* obj = &owner_id;
		obj->fromJson(value.dump());

    }

    const char *servicesKey = "services";
    
    if(object.has_key(servicesKey)) 
    {
        bourne::json value = object[servicesKey];


        

        AnyType* obj = &services;
		obj->fromJson(value.dump());

    }

    const char *emailKey = "email";
    
    if(object.has_key(emailKey)) 
    {
        bourne::json value = object[emailKey];


        
        jsonToValue(&email, value, "std::string");


    }

    const char *bot_typeKey = "bot_type";
    
    if(object.has_key(bot_typeKey)) 
    {
        bourne::json value = object[bot_typeKey];


        
        jsonToValue(&bot_type, value, "int");


    }

    const char *is_activeKey = "is_active";
    
    if(object.has_key(is_activeKey)) 
    {
        bourne::json value = object[is_activeKey];


        
        jsonToValue(&is_active, value, "bool");


    }


}

bourne::json
Bot::toJson()
{
    bourne::json object = bourne::json::object();

    




	object["user_id"] = getUserId().toJson();

    




	object["full_name"] = getFullName().toJson();

    




	object["api_key"] = getApiKey().toJson();

    




	object["default_sending_stream"] = getDefaultSendingStream().toJson();

    




	object["default_events_register_stream"] = getDefaultEventsRegisterStream().toJson();

    




	object["default_all_public_streams"] = getDefaultAllPublicStreams().toJson();

    




	object["avatar_url"] = getAvatarUrl().toJson();

    




	object["owner_id"] = getOwnerId().toJson();

    




	object["services"] = getServices().toJson();

    



    object["email"] = getEmail();


    



    object["bot_type"] = getBotType();


    



    object["is_active"] = isIsActive();



    return object;

}

AnyType
Bot::getUserId()
{
	return user_id;
}

void
Bot::setUserId(AnyType  user_id)
{
	this->user_id = user_id;
}

AnyType
Bot::getFullName()
{
	return full_name;
}

void
Bot::setFullName(AnyType  full_name)
{
	this->full_name = full_name;
}

AnyType
Bot::getApiKey()
{
	return api_key;
}

void
Bot::setApiKey(AnyType  api_key)
{
	this->api_key = api_key;
}

AnyType
Bot::getDefaultSendingStream()
{
	return default_sending_stream;
}

void
Bot::setDefaultSendingStream(AnyType  default_sending_stream)
{
	this->default_sending_stream = default_sending_stream;
}

AnyType
Bot::getDefaultEventsRegisterStream()
{
	return default_events_register_stream;
}

void
Bot::setDefaultEventsRegisterStream(AnyType  default_events_register_stream)
{
	this->default_events_register_stream = default_events_register_stream;
}

AnyType
Bot::getDefaultAllPublicStreams()
{
	return default_all_public_streams;
}

void
Bot::setDefaultAllPublicStreams(AnyType  default_all_public_streams)
{
	this->default_all_public_streams = default_all_public_streams;
}

AnyType
Bot::getAvatarUrl()
{
	return avatar_url;
}

void
Bot::setAvatarUrl(AnyType  avatar_url)
{
	this->avatar_url = avatar_url;
}

AnyType
Bot::getOwnerId()
{
	return owner_id;
}

void
Bot::setOwnerId(AnyType  owner_id)
{
	this->owner_id = owner_id;
}

AnyType
Bot::getServices()
{
	return services;
}

void
Bot::setServices(AnyType  services)
{
	this->services = services;
}

std::string
Bot::getEmail()
{
	return email;
}

void
Bot::setEmail(std::string  email)
{
	this->email = email;
}

int
Bot::getBotType()
{
	return bot_type;
}

void
Bot::setBotType(int  bot_type)
{
	this->bot_type = bot_type;
}

bool
Bot::isIsActive()
{
	return is_active;
}

void
Bot::setIsActive(bool  is_active)
{
	this->is_active = is_active;
}



